File: src/main/java/cat/xlagunas/andrtc/common/AuthenticationUtils.java
package cat.xlagunas.andrtc.common;

import cat.xlagunas.andrtc.user.UserDto;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;

public class AuthenticationUtils {
    public static long getPrincipalId(UsernamePasswordAuthenticationToken principal) {
        UserDto dto = (UserDto) principal.getPrincipal();
        return dto.id;
    }
}

File: src/main/java/cat/xlagunas/andrtc/common/AuthenticationUtils.kt
package cat.xlagunas.andrtc.common

import cat.xlagunas.andrtc.user.UserDto
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken

object AuthenticationUtils {
    fun getPrincipalId(principal: UsernamePasswordAuthenticationToken): Long {
        val dto = principal.principal as UserDto
        return dto.id
    }
}
